pkg/xds/envoy/listeners/v3/retry_configurer.go

```diff
@@ -93,6 +93,21 @@ func genHttpRetryPolicy(
 		policy.RetriableStatusCodes = conf.RetriableStatusCodes
 	}
 
+	for _, methodValue := range conf.GetRetriableMethods() {
+		if methodValue == mesh_proto.HttpMethod_NONE {
+			continue
+		}
```

Comment:
I agree that this check makes sense, but also add a validation check in `pkg/core/resources/apis/mesh/retry_validator.go` so users can get immediate feedback.